eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ThrowFromClientLoggerCheck.java

```diff
@@ -47,33 +47,9 @@
     public void visitToken(DetailAST token) {
         switch (token.getType()) {
             case TokenTypes.LITERAL_THROW:
-                DetailAST parentToken = token.getParent();
-                boolean throwsFromMethod = false;
-                while (parentToken != null) {
-                    int parentType = parentToken.getType();
-                    // Skip throw exception check if LITERAL_THROW from static method
-                    if (parentType == TokenTypes.METHOD_DEF) {
-                        DetailAST modifiersToken = parentToken.findFirstToken(TokenTypes.MODIFIERS);
-                        if (modifiersToken.branchContains(TokenTypes.LITERAL_STATIC)) {
-                            return;
-                        }
-                        throwsFromMethod = true;
-                    }
-                    // Skip throw exception check if LITERAL_THROW from static class
-                    if (parentType == TokenTypes.CLASS_DEF) {
-                        DetailAST modifiersToken = parentToken.findFirstToken(TokenTypes.MODIFIERS);
-                        if (modifiersToken.branchContains(TokenTypes.LITERAL_STATIC)) {
-                            return;
-                        }
-                        break;
-                    }
-                    // Skip throw exception check if LITERAL_THROW from constructor
-                    if (!throwsFromMethod && parentType == TokenTypes.CTOR_DEF) {
-                        return;
-                    }
-                    parentToken = parentToken.getParent();
+                if (!isValidScope(token)) {
```

Comment:
It feels like you are subverting the visitor pattern in isValidScope by doing the tree traversal yourself. You can save states in your logger.

1. When at a class def, is this class publicly visible?
2. When at a method def, is this method publicly visible?
3. I encountered a "throw", is it inside that method and class that is publicly visible(checked in previous steps)?  If so, then log.

Reply:
Yep. You are right. Will change to using the state of token.

Reply:
BTW, the logic will only check if the class is a static class because ClientLogger will not be a static variable, it will not able to use a non-static variable in a static class. The rule checks all main code package no matter if a class is public or private.